src/components/Tabs/TabPanels/TabPanels.jsx

```diff
@@ -1,27 +1,34 @@
-import React, { useRef, forwardRef } from "react";
+import React, { useRef, forwardRef, useMemo } from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
 import useMergeRefs from "../../../hooks/useMergeRefs";
 import "./TabPanels.scss";
 
-const TabPanels = forwardRef(({ className, id, activeTabId, animationDirection, children }, ref) => {
-  const componentRef = useRef(null);
-  const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
-
-  return (
-    <div ref={mergedRef} className={cx("tab-panels--wrapper", className)} id={id}>
-      {React.Children.map(children, (child, index) => {
+const TabPanels = forwardRef(
+  ({ className, id, activeTabId, animationDirection, children, renderOnlyActiveTab }, ref) => {
+    const componentRef = useRef(null);
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+    const renderedTabs = useMemo(() => {
+      return React.Children.map(children, (child, index) => {
         if (activeTabId === index) {
```

Comment:
we can do here  if (activeTabId === index || ! renderOnlyActiveTab
for reduce copies of the same code (the return line) and therefore it will be more simple.